### PR TITLE
Added GameState to published PlayInfo

### PR DESCRIPTION
**Pull request by MathewMacDougall**
### Description
Changes from RoboCup 2019

### Testing Done
unit tests still pass, validated with RoboCup gameplay

### Resolved Issues
None

### Length Justification
None

### CODE REVIEW

File: src/thunderbots/software/util/refbox_constants.cpp [hunk 1]
```diff
@@ -25,6 +25,18 @@ static const std::map<RefboxGameState, std::string> refbox_game_state_names = {
     {RefboxGameState::BALL_PLACEMENT_THEM, "BALL_PLACEMENT_THEM"},
     {RefboxGameState::REFBOX_GAME_STATE_COUNT, "REFBOX_GAME_STATE_COUNT"}};
 
+std::string name(const RefboxGameState& state)
+{
+    if (refbox_game_state_names.find(state) != refbox_game_state_names.end())
+    {
+        return refbox_game_state_names.find(state)->second;
+    }
```
Comment (garethellis0) [inline src/thunderbots/software/util/refbox_constants.cpp:34]: Nit: can avoid duplicate work:
```
    auto find_name_result = refbox_game_state_names.find(state);
    if (find_name_result != refbox_game_state_names.end())
    {
        return find_name_result->second;
    }
```

File: src/thunderbots/software/util/refbox_constants.cpp [hunk 2]
```diff
@@ -25,6 +25,18 @@ static const std::map<RefboxGameState, std::string> refbox_game_state_names = {
...
+    else
+    {
+        return "INVALID RefboxGameState";
```
Comment (garethellis0) [inline src/thunderbots/software/util/refbox_constants.cpp]: This seems like a pretty sub-ideal way to handle this case:
- the `RefboxGameState` isn't invalid, there just isn't an entry for it in the map
- we failed to find a name, but we're still successfully returning a string and relying on the user to see it propogated somewhere, trace it back here, and figure out the issue

I'd much rather we (in decreasing order of how nice I think the solution is):
- Refactor this enum to be a class hierarchy. So `GameState` would be an abstract class, and then `BallPlacementUs`, `BallPlacementThem`, `KickOffUs`, etc. would extend `GameState`. If you make each have a `getName()` function and make it pure virtual in `GameState`, we can enforce that every state must define it's own string
- return an optional
- throw an exception

File: src/thunderbots/software/util/refbox_constants.cpp [hunk 3]
```diff
@@ -25,6 +25,18 @@ static const std::map<RefboxGameState, std::string> refbox_game_state_names = {
     {RefboxGameState::BALL_PLACEMENT_THEM, "BALL_PLACEMENT_THEM"},
     {RefboxGameState::REFBOX_GAME_STATE_COUNT, "REFBOX_GAME_STATE_COUNT"}};
 
+std::string name(const RefboxGameState& state)
```
Comment (garethellis0) [inline src/thunderbots/software/util/refbox_constants.cpp:28]: Doesn't look like this is covered in any unit tests?

File: src/thunderbots/software/ai/hl/stp/stp.cpp [hunk 4]
```diff
@@ -238,7 +239,7 @@ std::optional<std::string> STP::getCurrentPlayName() const
 PlayInfo STP::getPlayInfo()
```
Comment (garethellis0) [inline src/thunderbots/software/ai/hl/stp/stp.cpp:239]: Doesn't look like we have test coverage for this function, a quick test here would at least sanity check it, and also the `name` function below.

**Review by garethellis0 — CHANGES_REQUESTED** [review]

File: src/thunderbots/software/util/refbox_constants.cpp (hunk 1 above)
Comment (MathewMacDougall) [reply src/thunderbots/software/util/refbox_constants.cpp:34]: done

Comment (garethellis0) [reply src/thunderbots/software/util/refbox_constants.cpp:34]: doesn't look like this is changed?

Comment (MathewMacDougall) [reply src/thunderbots/software/util/refbox_constants.cpp:34]: should be good now

File: src/thunderbots/software/util/refbox_constants.cpp (hunk 3 above)
Comment (MathewMacDougall) [reply src/thunderbots/software/util/refbox_constants.cpp:28]: added tests

File: src/thunderbots/software/util/refbox_constants.cpp (hunk 2 above)
Comment (MathewMacDougall) [reply src/thunderbots/software/util/refbox_constants.cpp]: now throws an exception

File: src/thunderbots/software/ai/hl/stp/stp.cpp (hunk 4 above)
Comment (MathewMacDougall) [reply src/thunderbots/software/ai/hl/stp/stp.cpp:239]: added unit tests

**Comment by MathewMacDougall** [comment]
@garethellis0 	ready for re-review

**Review by garethellis0 — APPROVED** [review]
LGTM :+1: :tada:

**Pull request merged**